public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -520,20 +522,29 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
     };
   }
 
+  createAnnotationQueryOptions = (options: any): DataQueryRequest<PromQuery> => {
+    const annotation = options.annotation;
+    const interval =
```

Comment:
Think we should use this.adjustInterval here so that the interval adjusts to time range (so the default 60s won't end up returning too many data points for a long time range).

Reply:
Not sure about that, isn't that covered in createQuery?

Reply:
Updated PR description so it's more clear what this fix does.